File: benchmarks/oopsla25/subjects/441.py
#!/usr/bin/env python3

import sys

def f441(s: str):
  if s[0] == "a":
    assert s[1] == "b"
    assert len(s) <= 2
  else:
    assert s[0] == "c"
    assert len(s) == 1

if __name__ == '__main__':
  my_string = ""
  if len(sys.argv) == 1:
    my_string = sys.stdin.read()
  else:
    fd = open(sys.argv[1], 'r')
    my_string = fd.read()
    fd.close()
  
  f441(my_string)

File: benchmarks/oopsla25/subjects/441.c
#include <assert.h>
#include <fcntl.h>
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <unistd.h>

void f441(const char *s) {
    if (s[0] == 'a') {
        assert(s[1] == 'b');
        assert(strlen(s) <= 2);
    } else {
        assert(s[0] == 'c');
        assert(strlen(s) == 1);
    }
}

int main(int argc, char *argv[]) {
  char my_string[10240] = {0};
  if (argc == 1) {
    read(fileno(stdin), my_string, 10240);
  } else {
    int fd = open(argv[1], O_RDONLY);
    read(fd, my_string, 10240);
    close(fd);
  }
  f441(my_string);
  return 0;
}
